### PR TITLE
feat: add meta-storage helper

### PR DESCRIPTION
**Pull request by sveisvei**
## JIRA Issue
[COREWEB-39](https://jira.finn.no/browse/COREWEB-39)

## Status
**READY**

## Description
* add helper class for writing and reading meta-data. 
* upgraded sinks with new versions

## Todos
- [x] Tests
- [x] Documentation

## Related PRs
* https://github.com/asset-pipe/asset-pipe-sink-gcs/pull/46
* https://github.com/asset-pipe/asset-pipe-sink-mem/pull/46
* https://github.com/asset-pipe/asset-pipe-sink-fs/pull/39

### CODE REVIEW

File: test/meta-storage.test.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+test('get() - should error if non-valid json', async () => {
+    const sink = new SinkMem();
+    const storage = new MetaStorage(sink);
+
+    sink.db[storage.getKey('key')] = 'non-valid-json';
+
+    expect(storage.get('key')).rejects.toEqual(
+        new SyntaxError('Unexpected token o in JSON at position 1')
+    );
```
Comment (digitalsadhu) [inline test/meta-storage.test.js]: Maybe it's worth handling invalid input in order to generate a better error message?

Comment (sveisvei) [reply test/meta-storage.test.js]: Agree

File: lib/meta-storage.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+module.exports = class MetaStorage {
+    constructor(sink) {
+        if (!sink) {
+            throw new Error('MetaStorage needs a sink');
+        }
```
[the remark this answers is not in the document]
Comment (sveisvei) [reply lib/meta-storage.js]: fixing

File: lib/meta-storage.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const assert = require('assert');
+
+module.exports = class MetaStorage {
+    constructor(sink) {
+        assert(sink, 'MetaStorage needs a sink');
```
Comment (trygve-lie) [inline lib/meta-storage.js]: Will this surface to the end user? If so, will the end user know what "MetaStorage" is?

File: lib/meta-storage.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    async get(fileName) {
+        const result = await this.sink.get(this.getKey(fileName));
+        if (result) {
+            try {
+                return JSON.parse(result);
+            } catch (e) {
+                throw new Error(
+                    `Failed parsing payload from key "${fileName}"`
```
Comment (trygve-lie) [inline lib/meta-storage.js]: What is "payload"?

Comment (SimenB) [reply lib/meta-storage.js]: Please keep the original error as well, so it's possible to see why it failed

File: lib/meta-storage.js (hunk 3 above)
Comment (sveisvei) [reply lib/meta-storage.js]: This is for internal usage as of now.

File: lib/meta-storage.js (hunk 4 above)
Comment (sveisvei) [reply lib/meta-storage.js]: Lets improve in new pr.

Comment (SimenB) [reply lib/meta-storage.js]: why?

Comment (sveisvei) [reply lib/meta-storage.js]: I have upcoming changes here soon, will add this.

**Pull request merged**